sentry-sidekiq/lib/sentry/sidekiq/error_handler.rb

```diff
@@ -5,7 +5,13 @@ module Sidekiq
     class ErrorHandler
       WITH_SIDEKIQ_7 = ::Gem::Version.new(::Sidekiq::VERSION) >= ::Gem::Version.new("7.0")
 
-      def call(ex, context)
+      # @param ex [Exception] the exception / error that occured
+      # @param context [Hash or Array] Sidekiq error context
+      # @param sidekiq_config [Sidekiq::Config] Sidekiq configuration,
+      #   defaults to Sidekiq's default configuration `Sidekiq.default_configuration`
+      #   Sidekiq will pass the config in starting Sidekiq 7.1.5, see
+      #   https://github.com/sidekiq/sidekiq/pull/6051
+      def call(ex, context, sidekiq_config = ::Sidekiq.default_configuration)
```

Comment:
I think `::Sidekiq.default_configuration` was introduced in 7.0 so this will break older Sidekiq versions. Also, we need to update `retry_limit` to use the passed `sidekiq_config` if its present.

Reply:
Hmmm. You're right — I think then we could set it to `nil` by default, but if it's present and has `retry_limit`, use that. I'll fix this up.

Reply:
Thank you again for pointing this out, and for the deferred cleanup of the env var. I think I got to it, not sure if my code is the most elegant.